cedar-policy-validator/src/types.rs

```diff
@@ -934,8 +1083,13 @@ impl EntityRecordKind {
     ) -> Option<EntityRecordKind> {
         use EntityRecordKind::*;
         match (rk0, rk1) {
-            (Record { attrs: attrs0 }, Record { attrs: attrs1 }) => Some(Record {
+            (Record { attrs: attrs0, .. }, Record { attrs: attrs1, .. }) => Some(Record {
                 attrs: Attributes::least_upper_bound(schema, attrs0, attrs1),
+                // We could do an explicit check here to see if `attr0 == attr1`
+                // to keep `false` for `open_attributes`, but this result
+                // is already achieved due to the subtype check done by
+                // `Type::least_upper_bound`.
+                open_attributes: true,
```

Comment:
I'm not sure why you don't want to be precise. If I have an `if/then/else` and both branches return a record literal, I'd like to know that the record literal is closed. That way I'll catch `(if * then { f: 1 } else { f: 2 }).g` in partial schema validation as an error.